rskj-core/src/main/java/co/rsk/peg/BridgeSupport.java

```diff
@@ -962,6 +962,86 @@ public void updateCollections(Transaction rskTx) throws IOException {
+
+        BtcTransaction svpFundTransactionUnsigned = new BtcTransaction(bridgeConstants.getBtcParams());
+        svpFundTransactionUnsigned.setVersion(BTC_TX_VERSION_2);
+
+        svpFundTransactionUnsigned.addOutput(spendableValueFromProposedFederation, proposedFederation.getAddress());
+        // complete tx with input and change output
+        SendRequest sendRequest = completeSvpSpendTransactionRequestToSend(svpFundTransactionUnsigned, feePerKb);
+        activeFederationWallet.completeTx(sendRequest);
+
+        return svpFundTransactionUnsigned;
+    }
+
+    private SendRequest completeSvpSpendTransactionRequestToSend(BtcTransaction transaction, Coin feePerKb) {
+        SendRequest sendRequest = SendRequest.forTx(transaction);
+        sendRequest.changeAddress = getActiveFederationAddress();
+        sendRequest.feePerKb = feePerKb;
+        sendRequest.missingSigsMode = Wallet.MissingSigsMode.USE_OP_ZERO;
+        sendRequest.recipientsPayFees = false;
+
+        return sendRequest;
+    }
+
+    private static Coin getSvpFundTxSentAmount(Federation proposedFederation, List<TransactionOutput> pegoutTransactionOutputs) {
+        // pegout value "requested" from proposed federation is the value actually sent
+        // since the active federation is paying for the fees
+
+        Script proposedFederationScriptPubKey = proposedFederation.getP2SHScript();
+        Optional<TransactionOutput> outputToProposedFederationOpt = pegoutTransactionOutputs.stream()
+            .filter(output -> output.getScriptPubKey().equals(proposedFederationScriptPubKey))
+            .findFirst();
+
+        if (!outputToProposedFederationOpt.isPresent()) {
+            String message = "The output for the proposed federation should be present in the svp fund transaction.";
+            logger.warn(message);
+            throw new IllegalStateException(message);
+        }
+        TransactionOutput outputToProposedFederation = outputToProposedFederationOpt.get();
+
+        return outputToProposedFederation.getValue();
+    }
+
+    private void performPegoutActions(BtcTransaction pegoutTransaction, Keccak256 rskTxHash, Coin requestedPegoutAmount) throws IOException {
+        addPegoutToPegoutsWaitingForConfirmations(pegoutTransaction, rskTxHash);
+        savePegoutTxSigHash(pegoutTransaction);
+        logReleaseRequested(rskTxHash.getBytes(), pegoutTransaction, requestedPegoutAmount);
+        logPegoutTransactionCreated(pegoutTransaction);
+    }
+
+    private void addPegoutToPegoutsWaitingForConfirmations(BtcTransaction pegoutTransaction, Keccak256 rskTxHash) throws IOException {
+        PegoutsWaitingForConfirmations pegoutsWaitingForConfirmations = provider.getPegoutsWaitingForConfirmations();
+        pegoutsWaitingForConfirmations.add(pegoutTransaction, rskExecutionBlock.getNumber(), rskTxHash);
+    }
+
+    private void logReleaseRequested(byte[] rskTxHashSerialized, BtcTransaction pegoutTransaction, Coin requestedPegoutAmount) {
```

Comment:
Why not receive the hash object here and serialize it in the method? Seems like that's a responsibility of this method